openpype/hooks/pre_mac_launch.py

```diff
@@ -0,0 +1,27 @@
+from openpype.lib import PreLaunchHook
+
+
+class LaunchWithTerminal(PreLaunchHook):
+    """Mac specific pre arguments for application.
+
+    Mac applications should be launched using "open" argument which is internal
+    callbacks to open executable. We also add argument "-an" to create new
+    process. This is used only for executables ending with ".app". It is
+    expected that these executables lead to app packages.
+    """
+    order = 1000
+
+    platforms = ["darwin"]
+
+    def execute(self):
+        executable = self.launch_context.executable
+        # Skip executables not starting with ".app"
+        if not executable.endswith(".app"):
```

Comment:
I am not sure if we shouldn't check here also for existence of executable and for `Path(executable).is_dir()`?

Reply:
> check here also for existence of executable

Existence of executable is first step of launching application (before launch hooks are discovered).

> and for Path(executable).is_dir()

The path with `.app` should be directory?

Reply:
application bundle on macos is directory with `.app` extensions and specific content (like `Content/MacOS` and `Resources` folders and `Info.plist` xml)

Reply:
Added simple `isdir` check so it should be fine now.